runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/Book.java

```diff
@@ -0,0 +1,147 @@
+	HOSIDIUS_LETTER(ItemID.HOSIDIUS_LETTER, "Hosidius Letter", "A letter from Lord Hosidius to the Council of Elders."),
+	WINTERTODT_PARABLE(ItemID.WINTERTODT_PARABLE, "Wintertodt Parable", "The Parable of the Wintertodt, by Anonymous."),
+	TWILL_ACCORD(ItemID.TWILL_ACCORD, "Twill Accord", "The Royal Accord of Twill."),
+	BYRNES_CORONATION_SPEECH(ItemID.BYRNES_CORONATION_SPEECH, "Byrnes Coronation Speech", "Speech of King Byrne I, on the occasion of his coronation."),
+	IDEOLOGY_OF_DARKNESS(ItemID.IDEOLOGY_OF_DARKNESS, "The Ideology of Darkness", "The Ideology of Darkness, by Philophaire."),
+	RADAS_JOURNEY(ItemID.RADAS_JOURNEY, "Rada's Journey", "The Journey of Rada, by Griselle."),
+	TRANSVERGENCE_THEORY(ItemID.TRANSVERGENCE_THEORY, "Transvergence Theory", "The Theory of Transvergence, by Amon Ducot."),
+	TRISTESSAS_TRAGEDY(ItemID.TRISTESSAS_TRAGEDY, "Tristessa's Tradgedy", "The Tragedy of Tristessa."),
+	TREACHERY_OF_ROYALTY(ItemID.TREACHERY_OF_ROYALTY, "The Treachery of Royalty", "The Treachery of Royalty, by Professor Answith."),
+	TRANSPORTATION_INCANTATIONS(ItemID.TRANSPORTATION_INCANTATIONS, "Transportation Incantations", "Transportation Incantations, by Amon Ducot."),
+	SOUL_JORUNEY(ItemID.SOUL_JOURNEY, "Soul Journey", "The Journey of Souls, by Aretha."),
+	VARLAMORE_ENVOY(ItemID.VARLAMORE_ENVOY, "Varlamore Envoy", "The Envoy to Varlamore, by Deryk Paulson.");
+
+	private static final Map<Integer, Book> BY_ID = buildById();
+
+	private static final Map<String, Book> BY_NAME = buildByName();
+
+	private static Map<Integer, Book> buildById()
+	{
+		HashMap<Integer, Book> byId = new HashMap<>();
+		for (Book b : Book.values())
+		{
+			byId.put(b.item, b);
+		}
+		return Collections.unmodifiableMap(byId);
+	}
+
+	private static final Map<String, Book> buildByName()
+	{
+		HashMap<String, Book> byName = new HashMap<>();
+		for (Book b : Book.values())
+		{
+			if (!b.isDarkManuscript)
+			{
+				byName.put(b.name, b);
+			}
+		}
+		return Collections.unmodifiableMap(byName);
+	}
+
+	public static Book byId(int id)
+	{
+		return BY_ID.get(id);
+	}
+
+	public static Book byName(String name)
+	{
+		return BY_NAME.get(name);
+	}
+
+	@Getter
+	private final int item;
+
+	@Getter
+	private final String name;
+
+	@Getter
+	private final String shortName;
+
+	@Getter
+	private final BufferedImage icon;
+
+	@Getter
+	private final boolean isDarkManuscript;
+
+	Book(int id, String shortName, String name)
+	{
+		this.item = id;
+		this.isDarkManuscript = false;
+		this.shortName = shortName;
+		this.name = name;
+		this.icon = getImage(Integer.toString(id));
+	}
+
+	Book(int id)
+	{
+		this.item = id;
+		this.isDarkManuscript = true;
+		this.name = "Dark Manuscript";
+		this.shortName = "Dark Manuscript";
+		this.icon = getImage(Integer.toString(id));
+	}
+
+	private static BufferedImage getImage(String name)
+	{
+		try
+		{
+			return ImageIO.read(Book.class.getResourceAsStream("items/" + name + ".png"));
```

Comment:
Does the existing itemmanager item images not work for this?

Reply:
They need to be used by the plugin panel, I was worried about racing with the client unless I added delay loading and synchronization. I figured this was much easier